### PR TITLE
ConfigMgr changes: Jwt_authn use per-route config, remove path_matcher

### PR DESCRIPTION
**Pull request by qiwzhang**
This is all golang changes in configManager:
1) jwt_authn to use per-route config;  
2) remove path_matcher filter

cache interface from envoy/go-control-plane is changed.  config_manager_test.go needs to change a lot to use the new API.

Next PR will remove path_match filter in c++ code

### CODE REVIEW

File: examples/testdata/path_matcher/envoy_config.json [hunk 1]
```diff
@@ -1,606 +0,0 @@
-{
```
Comment (nareddyt) [inline examples/testdata/path_matcher/envoy_config.json:1]: I would prefer to keep these files, maybe rename the directory to `route_match`. They show configs for snake-JSON variable bindings, autogenerated CORS, per-route configs with regex, etc. I find it helpful to reference.

File: src/go/bootstrap/static/bootstrap_test.go [hunk 2]
```diff
@@ -74,17 +74,6 @@ func TestServiceToBootstrapConfig(t *testing.T) {
 			serviceConfigPath: platform.GetFilePath(platform.DrServiceConfig),
 			envoyConfigPath:   platform.GetFilePath(platform.DrEnvoyConfig),
 		},
-		{
```
Comment (nareddyt) [inline src/go/bootstrap/static/bootstrap_test.go:77]: Ditto

File: src/go/configgenerator/route_generator.go [hunk 3]
```diff
@@ -201,6 +202,18 @@ func makePerRouteFilterConfig(operation string, method *configinfo.MethodInfo, h
...
 	}
+
+	// add JwtAuthn PerRouteConfig
+	if method.RequireAuth {
+		jwtPerRoute := &jwtpb.PerRouteConfig{
+			RequirementSpecifier: &jwtpb.PerRouteConfig_RequirementName{
+				RequirementName: operation,
+			},
+		}
+		jwt, _ := ptypes.MarshalAny(jwtPerRoute)
```
Comment (nareddyt) [inline src/go/configgenerator/route_generator.go:213]: nit: handle err

File: src/go/configmanager/config_manager_test.go [hunk 4]
```diff
@@ -269,13 +272,13 @@ func getListeners(configManager *ConfigManager, opts options.ConfigGeneratorOpti
 	if err != nil {
 		return nil, nil, "", err
 	}
-	resp := respInterface.(cache.Response)
+	resp, _ := respInterface.GetDiscoveryResponse()
```
Comment (nareddyt) [inline src/go/configmanager/config_manager_test.go:275]: nit: handle err

**Review by TAOXUY — APPROVED** [review]

File: examples/testdata/path_matcher/envoy_config.json (hunk 1 above)
Comment (qiwzhang) [reply examples/testdata/path_matcher/envoy_config.json:1]: Done

File: src/go/bootstrap/static/bootstrap_test.go (hunk 2 above)
Comment (qiwzhang) [reply src/go/bootstrap/static/bootstrap_test.go:77]: Done

File: src/go/configmanager/config_manager_test.go (hunk 4 above)
Comment (qiwzhang) [reply src/go/configmanager/config_manager_test.go]: Done

File: src/go/configgenerator/route_generator.go (hunk 3 above)
Comment (qiwzhang) [reply src/go/configgenerator/route_generator.go]: Done

**Review by TAOXUY — APPROVED** [review]

**Review by nareddyt — APPROVED** [review]

**Pull request merged**